mesa_expense_autopopulate/models/hr/expense.py

```diff
@@ -0,0 +1,132 @@
+from odoo import api, fields, models
+
+
+class Expense(models.Model):
+    _inherit = "hr.expense"
+
+    x_studio_field_Evfct = fields.Many2one(
+        "project.project",
+        string="Project",
+        compute="_compute_project",
+        inverse="_inverse_project",
+    )
+    team_id = fields.Many2one(compute="_compute_team", inverse="_inverse_team")
+    warehouse_id = fields.Many2one(
+        compute="_compute_warehouse", inverse="_inverse_warehouse"
+    )
+    user_id = fields.Many2one(compute="_compute_user", inverse="_inverse_user")
+    company_id = fields.Many2one(compute="_compute_company", inverse="_inverse_company")
+    # The following studio field is only included here because the database would otherwise not let this module install.
+    x_studio_aa_update = fields.Many2one("account.analytic.account")
+
+    @api.multi
+    @api.depends("sale_order_id")
+    def _compute_project(self):
+        """
+        Compute the Expense's Project. It defaults to the Expense's Sale Order's
+        Project.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.x_studio_field_Evfct = self.env["project.project"].search(
+                [("name", "=", expense_id.sale_order_id.x_studio_project)], limit=1
+            )
+
+    @api.multi
+    @api.depends("sale_order_id")
+    def _compute_team(self):
+        """
+        Compute the Expense's Sales Team. It defaults to the Expense's Sale
+        Order's Sales Team.
+
+        :return {NoneType}:
+        """
+        for expense_id in self:
+            expense_id.team_id = expense_id.sale_order_id.team_id
```

Comment:
This needs to happen on the rest of your computes with inverses as well, please.

Also, don't forget the `.id` after whatever you're assigning to the `Many2one` field.  Same on the below computes and inverses.  As long as it's assigning a record to a `Many2one`, then you should probably have it in there.  If it's a `Char`, `Integer`, etc you don't need it because it isn't holding another record's "link" (if you will.)

```suggestion
            expense_id.team_id = expense_id.force_team_id.id or expense_id.sale_order_id.team_id.id
```